### DISCUSSION TITLE
Unable to compile tasmota32, xdrv_01_9_webserver.ino (AsyncUDP.h not found)

### DISCUSSION THREAD

**Discussion by andreagiaquinto** [arendst/Tasmota#21198] [Support request]
Hello,
I download the latest update and I can't compile tasmota32.
Any suggestion?

In file included from D:/Progetti/Tasmota/Tasmota-dev/tasmota/tasmota_xdrv_driver/xdrv_01_9_webserver.ino:64:
C:/Users/andrea3/.platformio/packages/framework-arduinoespressif32/libraries/DNSServer/src/DNSServer.h:2:10: fatal error: AsyncUDP.h: No such file or directory       

******************************************************************
* Looking for AsyncUDP.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:AsyncUDP.h"
* Web  > https://registry.platformio.org/search?q=header:AsyncUDP.h
*
******************************************************************

    2 | #include <AsyncUDP.h>
      |          ^~~~~~~~~~~~
compilation terminated.
Retrieved `.pio\build\tasmota32\lib399\Network\NetworkClient.cpp.o' from cache
*** [.pio\build\tasmota32\src\tasmota.ino.cpp.o] Error 1
==================================================================== [FAILED] Took 48.34 seconds ====================================================================

**Comment by s-hadinger** [comment]
I had this. You need to update `platformio_tasmota_cenv.ini` and/or `platformio_override.ini`. I'm not sure what broke, but updating to the latest solved it.

**Reply by andreagiaquinto** [reply]
I did update to the latest and is still not working, same error as before.

**Reply by Jason2866** [reply]
Close VSC. Delete the whole hidden folder `.platformio` Do a fresh git clone in a different drive/folder.
Start VSC and "Open Folder" with the new cloned Tasmota. VSC/Platformio will reinstall everything.
With this working version you can now try and see what broke the older version.

**Comment by andreagiaquinto** [comment]
Ok I will try in that way, thank you.

**Reply by andreagiaquinto** [reply]
I did delete Close VSC, delete .platformio, clone in a different folder.
Now I have the following issues:

Linking .pio\build\tasmota32\firmware.elf
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans0.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans1.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans2.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans3.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans4.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans5.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans6.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans7.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans8.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans9.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans10.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans11.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans12.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans13.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans14.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans15.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans16.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans17.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans18.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans19.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans20.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans21.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans22.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans23.ltrans.o: No such file or directory
c:/users/andrea3/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/12.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find C:\Users\andrea3\AppData\Local\Temp\cccKzmzG.ltrans24.ltrans.o: No such file or directory
collect2.exe: error: ld returned 1 exit status
*** [.pio\build\tasmota32\firmware.elf] Error 1

**Comment by Jason2866** [comment]
Is/was the PC connected to the Internet after the deletion of the folder and when starting VSC?
The framework installation is faulty. Do not use Cloud Drives. Windows can't handle such long path names when compiling 
You have deleted the complete! folder `c:/users/andrea3/.platformio` before?

**Reply by andreagiaquinto** [reply]
Yes PC was connected to the internet.
I delete the complete c:/users/andrea3/.platformio folder before.

**Reply by andreagiaquinto** [reply]
I able to compile with success "tasmota" but not "tasmota32".

**Reply by andreagiaquinto** [reply]
Temporary I fixed the issues using ptatformio with debian 10.
There is something wrong with VSC installation, it was working until latest tasmota updates.

**Comment by gonzalop** [comment]
I'm seeing the same issue on debian sid. The development and 14.0.0 branches have the problem, while the 13.4.0 branch does not.

I tried replacing the "include <AsyncUDP.h>" with the contents of the header file and still got an error because of missing symbols. It seems that AsyncUDP.{cpp,h} are not being compiled.

**Comment by Jason2866** [comment]
The issue you encounter is caused from using an old setup.

**Reply by gonzalop** [reply]
I've deleted .pio and .platformio. docker-tasmota, which I use to build from source, downloads new packages before building. I'm going to check if there's something in docker-tasmota, but barring that, any ideas of what I might have that is old?

**Reply by Jason2866** [reply]
Check all `platformio*.ini` files. Probably the Docker image has not the latest.

**Reply by sfromis** [reply]
I'm seeing no build trouble with a local docker image built some weeks ago, which IIRC was due to trying if it helped with some build issue.

**Comment by gonzalop** [comment]
It was the platformio_override.ini in docker-tasmota that caused the problems.
Thanks!

**Comment by jeroenvermeulen** [comment]
Fix for me was:
```
mv  platformio_override.ini  platformio_override_old.ini
cp  platformio_override_sample.ini  platformio_override.ini
```

**Reply by sfromis** [reply]
The structure is that a "fresh" repository clone from Github will **not** have `platformio_override.ini`, but only `platformio_override_sample.ini`. If you do not create your own `platformio_override.ini` before first build, then `platformio_override_sample.ini` will be renamed. This is meant to facilitate having your own `platformio_override.ini` "survive" re-synchronizing Tasmota source code from Github, without overwriting your customized overrides.

As `docker-tasmota` does not come with any `platformio_override.ini`, whatever you had there must have come from your own prior activities, like doing builds. If you did that without providing your own `platformio_override.ini` first, the renaming will happen as mentioned above, which can be "quite inconvenient" if you intended to inject your own overrides.
